drivers/node/vsphere/vsphere.go

```diff
@@ -862,3 +862,336 @@ func (v *vsphere) RemoveNonRootDisks(n node.Node) error {
+	if err != nil {
+		return fmt.Errorf("error initiating VM relocate: %v", err)
+	}
+	return task.Wait(ctx)
+}
+
+// FindVMByName finds a virtual machine by its name.
+func (v *vsphere) FindVMByName(vmName string) (*object.VirtualMachine, error) {
+	log.Infof("VM Name is: %v", vmName)
+	finder, err := v.getVMFinder()
+	if err != nil {
+		return nil, err
+	}
+	vm, err := finder.VirtualMachine(v.ctx, vmName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find VM: %v", err)
+	}
+	return vm, nil
+}
+
+// FindDatastoreByName finds a datastore by its name.
+func (v *vsphere) FindDatastoreByName(dsName string) (*object.Datastore, error) {
+	finder, err := v.getVMFinder()
+	if err != nil {
+		return nil, err
+	}
+	ds, err := finder.Datastore(v.ctx, dsName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find datastore: %v", err)
+	}
+	return ds, nil
+}
+
+// FindVMByIP finds the vsphere Name of a VM through its Data IP Address
+func (v *vsphere) FindVMByIP(node node.Node) (*object.VirtualMachine, error) {
+	log.Infof("Searching VM by IP Addresses: %v", node.Addresses)
+	finder, err := v.getVMFinder()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get VM finder: %v", err)
+	}
+
+	vms, err := finder.VirtualMachineList(v.ctx, "*")
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve VM list: %v", err)
+	}
+
+	for _, vm := range vms {
+		var vmProps mo.VirtualMachine
+		err = vm.Properties(v.ctx, vm.Reference(), []string{"guest.net"}, &vmProps)
+		if err != nil {
+			log.Infof("failed to get properties for VM: %s, error: %v", vm.Name(), err)
+			continue
+		}
+
+		for _, net := range vmProps.Guest.Net {
+			for _, ip := range net.IpAddress {
+				for _, nodeIP := range node.Addresses {
+					if ip == nodeIP {
+						log.Infof("Found VM by IP Address %s: %s", ip, vm.Name())
+						return vm, nil
+					}
+				}
+			}
+		}
+	}
+	return nil, fmt.Errorf("no VM found with the given IP addresses: %v", node.Addresses)
+}
+
+// ValidateDatastoreUpdate validates the cloud drive configmap after Storage vmotion is done
+func (v *vsphere) ValidateDatastoreUpdate(preData, postData map[string]DriveSet, nodeUUID string, targetDatastoreID string) bool {
+	preNodeData, preExists := preData[nodeUUID]
+	postNodeData, postExists := postData[nodeUUID]
+
+	if !preExists || !postExists {
+		return false
+	}
+
+	postDiskDatastores := make(map[string]string)
+	for _, postDrive := range postNodeData.Configs {
+		postDiskDatastores[postDrive.DiskUUID] = postDrive.Labels["datastore"]
+	}
+
+	allMoved := true
+	for _, preDrive := range preNodeData.Configs {
+		postDSName, exists := postDiskDatastores[preDrive.DiskUUID]
+		if !exists {
+			log.Infof("No post-migration data found for disk with UUID %v", preDrive.DiskUUID)
+			return false
+		}
+
+		postDS, err := v.FindDatastoreByName(postDSName)
+		if err != nil {
+			log.Errorf("Failed to find datastore with name %v: %v", postDSName, err)
+			return false
+		}
+		postDSID := postDS.Reference().Value
+
```

Comment:
You should check for dastostore of targetDatastoreID should be compared with postDSName here